AirCasting/SearchAndFollow/CompleteScreen/Views/SearchAndFollowChartViewModel.swift

```diff
@@ -21,8 +21,19 @@ class SearchAndFollowChartViewModel: ObservableObject {
     func generateEntries(with measurements: [ChartMeasurement], thresholds: ThresholdsValue) -> (Date?, Date?) {
         var times: [Date] = []
         var buffer: [ChartMeasurement] = []
+        var updatedMeasurements = measurements
         
-        for measurement in measurements.reversed() {
+        // This allows us to delete all of the measurements that cannot fulfill the whole hour range.
+        // Result of this, is the chart that is the same; as on the session card.
+        guard let firstElement = updatedMeasurements.reversed().first?.time else { return (nil, nil) }
+        let hoursForRemoval = Calendar.current.component(.hour, from: firstElement)
+        for (index, item) in updatedMeasurements.enumerated().reversed() {
+            if Calendar.current.component(.hour, from: item.time) == hoursForRemoval {
+                updatedMeasurements.remove(at: index)
+            }
+        }
+        
+        for measurement in updatedMeasurements.reversed() {
```

Comment:
What do you think about adding this check `if Calendar.current.component(.hour, from: item.time) == hoursForRemoval` below this line inside this loop, and if it is true then continuing to the next measurement with `continue`? It will remove this additional loop through all of the measurements.

Reply:
Really like the idea. Changed!

Reply:
I meant that we can do it like this:
<img width="956" alt="Screenshot 2022-05-12 at 10 22 44" src="https://user-images.githubusercontent.com/15946812/168025884-aca52518-3fba-4fb4-9a4e-f9a6e797916f.png">

However, this only works well for Airbeam sessions, because for OpenAQ data we often don't have measurement from most recent hour and for example now, when there is 4:00 in New York, the lates measurement for one session is from 1:00. So we actually want to compare the hour of the measurement to the current time.